src/pool/srv_pool.c

```diff
@@ -6863,6 +6863,15 @@ pool_svc_update_map(struct pool_svc *svc, crt_opcode_t opc, bool exclude_rank,
 	daos_epoch_t			rebuild_eph = d_hlc_get();
 	uint64_t			delay = 2;
 
+	/*
+	 * For simplicity, deny drain and extend operations in
+	 * the no_data_sync reinteration mode.
+	 */
+	if (svc->ps_pool->sp_reint_mode == DAOS_REINT_MODE_NO_DATA_SYNC) {
+		if (opc != MAP_REINT && opc != MAP_EXCLUDE)
```

Comment:
maybe for future patch? add a test so that with no_data_sync mode and a pool extend or pool drain is attempted, this is confirmed to be returned to the client/test program?